### ISSUE TITLE
Conversion error when using the 'grid' package

### ISSUE THREAD

**Issue by hogend** [rpy2/rpy2#804]
**Description**
The following code demonstrates the issue
```python
import numpy as np

import rpy2.robjects as robjects

from rpy2 import __version_vector__ as rpy2_version
from rpy2.robjects import numpy2ri, FloatVector
from rpy2.robjects.lib import grid

numpy2ri.activate()
grid.activate()

if __name__ == '__main__':
    print(rpy2_version)

    def pretty(vx):
        return np.asarray(robjects.r('grid.pretty')(FloatVector(vx)))

    print(pretty(np.array([1.0, 4.0])))
```
This will generate an exception with rpy2 3.4.3:
```
(3, 4, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/hogend/Projects/KGT/scripts/tt.py", line 18, in <module>
    print(pretty(np.array([1.0, 4.0])))
  File "/home/hogend/Projects/KGT/scripts/tt.py", line 16, in pretty
    return np.asarray(robjects.r('grid.pretty')(FloatVector(vx)))
  File "/usr/lib/python3/dist-packages/rpy2/robjects/functions.py", line 198, in __call__
    return (super(SignatureTranslatedFunction, self)
  File "/usr/lib/python3/dist-packages/rpy2/robjects/functions.py", line 126, in __call__
    res = conversion.rpy2py(res)
  File "/usr/lib/python3/dist-packages/rpy2/robjects/lib/grid.py", line 269, in grid_rpy2py
    rcls = pyobj.rclass
AttributeError: 'numpy.ndarray' object has no attribute 'rclass'
```

**Expected behavior**
The expected output would be:
```
(3, 4, 3)
[1.  1.5 2.  2.5 3.  3.5 4. ]
```

**Error**
See above

**Additional context**
The problem disappears if `grid_rpy2py` in `robjects/lib/grid.py` is changed from
```python
def grid_rpy2py(robj):

    pyobj = original_rpy2py(robj)

    if not isinstance(pyobj, robjects.RS4):
        rcls = pyobj.rclass
        if rcls is NULL:
            rcls = (None, )
        try:
            cls = _grid_dict[rcls[0]]
            pyobj = cls(pyobj)
        except KeyError:
            pass

    return pyobj
```
to
```python
def grid_rpy2py(robj):

    pyobj = original_rpy2py(robj)

    if not isinstance(pyobj, robjects.RS4):
        rcls = getattr(pyobj, 'rclass', NULL)
        if rcls is NULL:
            rcls = (None, )
        try:
            cls = _grid_dict[rcls[0]]
            pyobj = cls(pyobj)
        except KeyError:
            pass

    return pyobj
```

**Comment by lgautier** [comment]
Thanks for the report. This problem is indeed with the converted in grid.

Note that the use of the methods `activate` is discouraged, they are deprecated, and they will be gone in the next major release.

**Comment by lgautier** [comment]
There was a quite of refactoring involved but the fix in now in the main branch. It will be included in release 3.5.0.
